### ISSUE TITLE
pygame.camera doesn't work on windows

### ISSUE THREAD

**Issue by illume** [pygame/pygame#160]
Originally reported by: **Bartek Kowalski (Bitbucket: [netrick](https://bitbucket.org/netrick), GitHub: Unknown)**

----------------------------------------

I tried both python 2.7 and 3.3. On windows, when I try to run anything using camera module I get:

  File "E:\Python33\lib\site-packages\pygame\camera.py", line 42, in init
    from pygame import _camera
ImportError: cannot import name _camera

----------------------------------------
- Bitbucket: https://bitbucket.org/pygame/pygame/issue/160

**Comment by feifeimao666** [comment]
I use python3.6  and pygame 1.9.3 on windows,but it still doesn't work!

**Comment by imwasenabled** [comment]
In reviewing the code in camera.py I discovered three small issues and a fix for this problem.

The first issue is that there is no _camera.py or _camera.pyd for Windows only for Linux and Mac (old camera) and as a result on line 43 "from pygame import _camera" fails because neither the .py or .pyd exist.

The second issue, related to issue #1 above, is that there is a "use__camera" boolean flag in the code but by default it is set to True and not set to False for Windows.  It should be set to False when sys.platform is detected as 'win32' on line 17 of camera.py.

The third issue and the source of this initial bug report is that camera.py on line 42 attempts to import _camera and fails because it does not exist but all that is necessary is to move the "if use__camera:"  conditional up three lines from line 45 so we first test if we should even be importing _camera, which we shouldn't on Windows, and if not then we skip this import request.

So my fix is to make two small changes.  First add the setting of use__camera to False after line 18:
[current]
17    if sys.platform == 'win32':
18        use_vidcapture = True
19
20    elif "linux" in sys.platform:

[new]
17    if sys.platform == 'win32':
18        use_vidcapture = True
19        use__camera = False
20
21    elif "linux" in sys.platform:

Second,  move the import of _camera inside the conditional test for use__camera on line 45.
[current - after making use__camera edit above]
41    # the _camera module has some code which can be reused by other modules.
42    #  it will also be the default one.
43    from pygame import _camera
44    colorspace = _camera.colorspace
45
46    if use__camera:
47        list_cameras = _camera.list_cameras

[new]
41    # the _camera module has some code which can be reused by other modules.
42    #  it will also be the default one.
43    if use__camera:
44        from pygame import _camera
45        colorspace = _camera.colorspace
46
47        list_cameras = _camera.list_cameras

Finally, as was pointed out on the Windows platform the VideoCapture dependency must be installed.

**Comment by fabiolouco** [comment]
> VideoCapture‑0.9.5‑cp36‑cp36m‑win_amd64.whl

Thank you very much, it worked like a charm!

**Comment by Srivallidevi** [comment]
> In reviewing the code in camera.py I discovered three small issues and a fix for this problem.
> 
> The first issue is that there is no _camera.py or _camera.pyd for Windows only for Linux and Mac (old camera) and as a result on line 43 "from pygame import _camera" fails because neither the .py or .pyd exist.
> 
> The second issue, related to issue #1 above, is that there is a "use__camera" boolean flag in the code but by default it is set to True and not set to False for Windows. It should be set to False when sys.platform is detected as 'win32' on line 17 of camera.py.
> 
> The third issue and the source of this initial bug report is that camera.py on line 42 attempts to import _camera and fails because it does not exist but all that is necessary is to move the "if use__camera:" conditional up three lines from line 45 so we first test if we should even be importing _camera, which we shouldn't on Windows, and if not then we skip this import request.
> 
> So my fix is to make two small changes. First add the setting of use__camera to False after line 18:
> [current]
> 17 if sys.platform == 'win32':
> 18 use_vidcapture = True
> 19
> 20 elif "linux" in sys.platform:
> 
> [new]
> 17 if sys.platform == 'win32':
> 18 use_vidcapture = True
> 19 use__camera = False
> 20
> 21 elif "linux" in sys.platform:
> 
> Second, move the import of _camera inside the conditional test for use__camera on line 45.
> [current - after making use__camera edit above]
> 41 # the _camera module has some code which can be reused by other modules.
> 42 # it will also be the default one.
> 43 from pygame import _camera
> 44 colorspace = _camera.colorspace
> 45
> 46 if use__camera:
> 47 list_cameras = _camera.list_cameras
> 
> [new]
> 41 # the _camera module has some code which can be reused by other modules.
> 42 # it will also be the default one.
> 43 if use__camera:
> 44 from pygame import _camera
> 45 colorspace = _camera.colorspace
> 46
> 47 list_cameras = _camera.list_cameras
> 
> Finally, as was pointed out on the Windows platform the VideoCapture dependency must be installed by visiting https://www.lfd.uci.edu/~gohlke/pythonlibs/#videocapture (as of today, 2018-09-08), downloading it locally and then using pip to install it like this: pip install [path-to-file]\VideoCapture‑0.9.5‑cp36‑cp36m‑win_amd64.whl (this is for the 64-bit version for Python 3.6). Note that I had to download the correct .whl to match my version of Python and the architecture installed (win32 or amd64).

After modifying the code as above, my camera is running continuosly , tried even task manager to stop it, but still laptop's inbuilt  camera's light is on.

**Comment by MyreMylar** [comment]
The camera module does work on windows in pygame 2 with python 3.7, using a Logitech 1080P webcam. It also works in Pygame 1.9.5 & 1.9.6 with Python 3.6.

code:

```
import pygame.camera

pygame.init()
pygame.camera.init()
cam_list = pygame.camera.list_cameras()
cam = None
if cam_list:
    cam = pygame.camera.Camera(cam_list[0], (640, 480))
    cam.start()

window_surface = pygame.display.set_mode((640, 480))

is_running = True
while is_running:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            is_running = False

    if cam is not None:
        image = cam.get_image()
        window_surface.blit(image, (0, 0))

    pygame.display.update()
```

using VideoCapture for the backend downloaded from here:
https://www.lfd.uci.edu/~gohlke/pythonlibs/#videocapture

There were no issues with the camera light staying on after the program was closed.

**Comment by MyreMylar** [comment]
Related:

https://github.com/pygame/pygame/issues/1741

**Comment by MyreMylar** [comment]
The issues prevent this from being closed are now resolved and merged and it should be fixed in 2.0.0.dev10.

**Comment by giohappy** [comment]
camera working on Windows 11 until version `2.5.2` with `Python 3.11`, but stopped working since version `2.6.0`
